# Genetic Diseases and calculating risk

## Distignuishing between Monogenic and Multifactorial Genetic Diseases

### Penetrance and Phenocopy

1. **Penetrance**:

- Penetrance is classified as the **proportion of individuals** (within a group) who are carrying a **specific variant** and also **present an associated phenotype** (trait) with that variant.

- It can only be calculated by **comparing genomic data between family members**, NOT large populations.

Or in other words, (in our case), how many people present an allelic variant and trait **within a family**. 

Penentrance is measured with the following formula:

==Penetrance==
: $ [Y] / X $

Where **X**:

Refers to the number of people with the "malicious" genotype (**all cariers within a family**-*affected or not*)

Where **Y**:

Refers to the number of people who have been **affected/have the traits** of the **"malicious"** phenotype

A **HIGH** level of penetrance is set on a threshold between ==**0.8 < 1**==.

2. **Phenocopy**:

- Could be seen as the opposite of penetrance;

Phenocopy
: $[X] / Y$

Where **X**:

Refers to the number of people who have been **affected/have the traits** of the **"malicious"** phenotype

Where **Y**:

Refers to the number of people who **are healthy** and have a **completely different genotype** to those with disease.

### Mendelian Entity

A **rare variant**, which is ==associated with **high** disease risk==.

When the **penetrance** of a specific variant **exceeds 0.8**, these are typically seen in **Monogenic** diseases.

### Susceptibility Genes

Are often **frequent variants**, which are thereby associated with **low disease-risk**. 

- If these variants were actually high-disease risk and frequent, we would all be dead.

### Monogenic Diseases: Hereditary

- Characterised by a **single genetic problem**; a **Mendelian** entity. As such, these diseases are ==**NOT characterised** by **environmental factors**==, their impact is **low**.

- Characterised by ***rare disease frequency*** within the larger population, but ***inversely*** within families, due to **high** penetrance, are **NOT** rare.

- While there is a **single driving genetic problem**, consider that *mutations will still occur in other genes*; they're just **not causal**.

- Keeping all of the above in mind, we can say that **familial agregation** is **very high**, as the **penetrance within the family** is high.

### Complex Diseases: Multifactorial

- Characterised by a **range of genetic factors**; both **genetic and environmental**. Primarily, **environmental factors play a medium to highly important role** in the development of these diseases.

Consider the effect of environmental factors in multifactoral diseases as adding petrol to a fire.

Multifactoral diseases are **on a spectrum**, where the **effect environmental factors** have on the individual, greatly depend on the allelic genotype of the individual.

- Characterised by **low** penetrance within families and by extension, ==**multiple confounding genes**==, **confounding factors**, which therfore mean **low familial aggragation**. (Consider that **~5-10% of families have medium-high penetrance** with multifactorial disease)

### Example: Monogenic

**PHENYLKETONURIA**
- Inability to properly convert phenylalanine (one of the amino acids found in food) into tyrosine
- Accumulation of toxic compounds for the brain and significant mental retardation from childhood
- Incidence :  1/16000
- **Mendelian Identity**: Gene PAH (phenylalanine hydroxylase) - chrom.12 – recessive trans. 
- ~ 500 mutations identified
- Mut. cause enzyme deficiency
- High penetrance
- if dietary treatment at birth --> penetrance = 0

### Example: Multifactorial

**BREAST CANCER**
(While some women observe high penetration in their family, we do not know what the mendelian entity is to drive this inheritance in those extreme cases.)

## Genetic Markers

- **SNVs**: DNA sequence variation in which a single nucleotide — A, T, C or G — differs between members of the same species.
- **SNP = single nucleotide polymorphism**: SNV occurring commonly within a population(> 1%).
- **Large copy number variants (CNVs)>1 kb**
- **Small insertions/deletions (indels)**

## What do you do with genetic markers?

Functional genomics/Pharmaceuticals
: Use genetic markers such as SNPs, to **observe their effect** on the ***function*** of "vitality" genes. Additionally, identify interactions between genetic and environmental factors to generate therapeutic targets for medicine.

Personalised Medicine/Counseling
: Take for example counseling for pregnant women, where **considering familial history of disease** (monogenic diseases) they can recommend preventative treatment once the baby has been born; take for example, dietary treatment at birth --> penetrance = 0 in PHENYLKETONURIA.

### Preferential markers: SNPs

- Most commonly used due to their **easier identification and isolation within samples**. 
- Are evenly distributed across the sequenced genome, making them easier to identify based on patterns.

## GWAS

- Genotyping of 500,000 – 1 million SNPs covering the genome in thousands of cases / controls.
- Genome to genome comparative studies, to locate control and case variants.

## Positional cloning

- ***Linkage analyses***: 