## Multiple Sequence Alignment

### Looking at an alignment

![](<Screenshot 2024-10-22 at 08.55.56.png>)

Concerning the start of the protein sequences aligned, one could *hypothesise that the start M*, is **missing** from the *first three sequences*, or it the *following A* has been **converted into M and the M was lost to deleterious SNV events**.

What appears as **protein truncation** at the last aligned sequence, *concerning its frameshift*, the ***best explanation is that its possibly misaligned***; therefore the alignment depicted is poor. 

Different tools will be more powerful for homologous sequences, or different parameters can help account for the condition of your sequences (poor sequencing, low homozygosity etc.).

### 