# Framework for notes

## What are motifs?

Sequence motifs have an important role in understanding protein structure and function, genetic regulations, and protein-protein or protein-nucleotide interactions.

When it comes to proteins, the term “motif” has a slightly different definition. The motif is a structural element found in all members of a protein family. It contains residues essential to a conserved function, not necessarily consecutive (this is a "pattern"), but quite close in the 3D structure, because they participate in the same function (active site, etc.). A pattern does not have its own folding.

**The pattern** is a degenerate sequence and/or composed of different patterns (in the strict sense) which can be separated by variable regions.

## Benefits of searching for patterns

- Determination of the function of a new sequence (identification of functional elements)
- Identification of open phases (search for specific associated signals)
- Establishment of a restriction map (search for cutting sites recognized by restriction enzymes)
- Search in a set of sequences of the same “signature” capable of characterizing them (identification of a protein family)

The determination of **specific motifs** generally comes from the **analysis of a multiple alignment**.

## Motifs in DNA or RNA sequences:

- Promoter regions (e.g. TATA box), transcription factor binding sites, enzymatic restriction sites, splicing signals

Before characterising motifs, a multiple sequence alignment is needed.

Take the following figure for analysis:

![alt text](<Screenshot 2024-11-05 at 10.44.34.png>)

## Patterns to find in proteinic sequences:

# RAUL SUUPLEMENT LITTLE BITCH

# ADD SHIT RAUL

# SLAVE

